### PR TITLE
added position NaN check in flycam moving reducer and url manager

### LINKED ISSUE #3432: Prevent position values from becoming NaN
I don't know how this can be provoked from within wk, but just clicking at the following link should not break wk:

https://webknossos.brain.mpg.de/annotations/Explorational/5b870c794e00006605e8ae87#NaN,NaN,NaN,0,4.13,405

Ideally, we would have a hard constraint that the position values cannot become NaN. Maybe in the position setting reducer?
For the URL case, the url state manager should probably do that.

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR adds a NaN check to all position changes in the flycam movement reducer. 
Nan values encoded in the URL are treated as 0s.

### URL of deployed dev instance (used for testing):
- [https://preventnonnumericpositionvalues.webknossos.xyz](https://preventnonnumericpositionvalues.webknossos.xyz)
### Steps to test:
- How to reproduce a NaN position calculation is unknown.
- But everything should work normally
- opening a tracing with: "...annotations/Explorational/someid#NaN,NaN,NaN,0,2.00" should not crash. It should set the initial position to 0,0,0.

### Issues:
- fixes #3432

------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/oxalis/controller/url_manager.js [hunk 1]
```diff
@@ -65,7 +65,8 @@ class UrlManager {
     if (stateString) {
       const stateArray = stateString.split(",").map(Number);
       if (stateArray.length >= 5) {
-        state.position = Utils.numberArrayToVector3(stateArray.slice(0, 3));
+        const positionValues = stateArray.slice(0, 3).map(value => (!isNaN(value) ? value : 0));
```
Comment (daniel-wer) [inline app/assets/javascripts/oxalis/controller/url_manager.js]: I would suggest to move the .map with the isNaN check two lines up so it is executed for all the numbers in the stateArray :) Currently, setting, for example, the zoomstep to a string value in the url hash results in an error as well.

File: app/assets/javascripts/oxalis/model/reducers/flycam_reducer.js [hunk 2]
```diff
@@ -171,13 +173,13 @@ function FlycamReducer(state: OxalisState, action: Action): OxalisState {
     case "SET_POSITION": {
       // cannot use M4x4.clone because of immutable-seamless
       const matrix = cloneMatrix(state.flycam.currentMatrix);
-      if (action.dimensionToSkip !== 0) {
+      if (action.dimensionToSkip !== 0 && !isNaN(matrix[12])) {
```
Comment (daniel-wer) [inline app/assets/javascripts/oxalis/model/reducers/flycam_reducer.js]: I think you're checking the wrong value here. `matrix` is from the store and action.position should be checked.
```suggestion
      const { position } = action;
      if (action.dimensionToSkip !== 0 && !isNaN(position[0])) {
```

Comment (daniel-wer) [reply app/assets/javascripts/oxalis/model/reducers/flycam_reducer.js]: Same for the next two checks.

File: app/assets/javascripts/oxalis/model/reducers/flycam_reducer.js [hunk 3]
```diff
@@ -191,12 +193,18 @@ function FlycamReducer(state: OxalisState, action: Action): OxalisState {
       return setDirectionReducer(state, action.direction);
     }
 
-    case "MOVE_FLYCAM":
+    case "MOVE_FLYCAM": {
+      let newMatrix = M4x4.translate(action.vector, state.flycam.currentMatrix, []);
+      if (newMatrix.includes(NaN)) {
+        // if there was a miscalculation, do not move the matrix
+        newMatrix = state.flycam.currentMatrix;
```
Comment (daniel-wer) [inline app/assets/javascripts/oxalis/model/reducers/flycam_reducer.js]: You can also simply early-return the old state here and make `newMatrix`const again:
```suggestion
        return state;
```

Comment (daniel-wer) [reply app/assets/javascripts/oxalis/model/reducers/flycam_reducer.js]: Or alternatively, revert the changes in this `case` and check the action.vector for NaN in the same way as you do in the `moveReducer` :)

`if (action.vector.includes(NaN)) return state;`

**Comment by MichaelBuessemeyer** [comment]
@daniel-wer I think this PR is ready for another review.

**Pull request merged**